validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -287,7 +286,8 @@ def __init__(self, squash_handler, first_state_hash, always_persist):
         # Transaction results
         self._txn_results = {}
 
-        self._transactions = OrderedDict()
+        self._txns_available = []
```

Comment:
This seems to be an addition of more parallel arrays/dictionaries, with respect to knowledge of the state a transaction as it moves through the schedule.  We should consider a refactor that collects the information about the state of a transaction (available, results etc) into a single object and a single store (either an array or map) of the transactions.  Any thing else is just a view of this data.

Reply:
That sounds like a good idea.